fuel-txpool/src/service.rs

```diff
@@ -111,13 +176,37 @@ impl Context {
 
         loop {
             tokio::select! {
-                event = self.receiver.recv() => {
-                    if matches!(event,Some(TxPoolMpsc::Stop) | None) {
+                new_transaction = self.incoming_tx_receiver.recv() => {
+                    if new_transaction.is_err() {
+                        error!("Incoming tx receiver channel closed unexpectedly; shutting down transaction pool service.");
                         break;
                     }
-                    let broadcast = self.broadcast.clone();
+
+                    let txpool = txpool.clone();
                     let db = self.db.clone();
+                    let tx_status_sender = self.tx_status_sender.clone();
+
+                    tokio::spawn( async move {
+                        let txpool = txpool.as_ref();
+                        match new_transaction.unwrap() {
+                            TransactionBroadcast::NewTransaction ( tx ) => {
+                                let txs = vec!(Arc::new(tx));
+                                TxPool::insert(txpool, db.as_ref().as_ref(), tx_status_sender, txs).await
```

Comment:
This shouldn't be done unless we change the current gossip logic to only broadcast to immediate peers (depth 1). Right now, any gossiped messages will be propagated to the whole network. We have other P2P tasks to investigate how to prevent nodes from re-gossiping invalid data.

Reply:
https://github.com/FuelLabs/fuel-core/issues/595

Reply:
Yes, it may not be clear from a cold read, but this branch is executed when new transactions are received from gossip. In this case, we do not want to echo the same transaction back to the network with a broadcast.